packages/cli/lib/sandboxes.js

```diff
@@ -66,25 +80,51 @@ async function getAvailableSyncTypes(parentAccountConfig, config) {
   return syncTypes.map(t => ({ type: t.name }));
 }
 
-const sandboxCreatePersonalAccessKeyFlow = async (env, account, name) => {
-  const configData = await personalAccessKeyPrompt({ env, account });
+/**
+ * @param {String} env - Environment (QA/Prod)
+ * @param {Object} result - Sandbox instance returned from API
+ * @param {Boolean} force - Force flag to skip prompt
+ * @returns {String} validName saved into config
+ */
+const saveSandboxToConfig = async (env, result, force = false) => {
+  // const configData = { env, personalAccessKey: result.personalAccessKey };
+  // TODO: Temporary, remove
+  const configData = await personalAccessKeyPrompt({
+    env,
+    account: result.sandbox.sandboxHubId,
+  });
+  // End temporary section
   const updatedConfig = await updateConfigWithPersonalAccessKey(configData);
-
   if (!updatedConfig) {
-    process.exit(EXIT_CODES.SUCCESS);
+    throw new Error('Failed to update config with personal access key.');
   }
 
   let validName = updatedConfig.name;
-
   if (!updatedConfig.name) {
-    const nameForConfig = name
+    const nameForConfig = result.sandbox.name
       .toLowerCase()
       .split(' ')
       .join('-');
-    const { name: promptName } = await enterAccountNamePrompt(nameForConfig);
-    validName = promptName;
+    validName = nameForConfig;
+    const invalidAccountName = accountNameExistsInConfig(nameForConfig);
+    if (invalidAccountName) {
+      if (!force) {
+        logger.log(
+          i18n(
+            `cli.lib.prompts.enterAccountNamePrompt.errors.accountNameExists`,
+            { name: nameForConfig }
+          )
+        );
+        const { name: promptName } = await enterAccountNamePrompt(
+          nameForConfig
+        );
+        validName = promptName;
+      } else {
+        // Basic invalid name handling when force flag is passed
+        validName = nameForConfig + Math.floor(Math.random() * 10);
```

Comment:
This could theoretically still result in a conflict though right? If the user runs this multiple times. Would something like a timestamp work better?

Reply:
ah yeah good call, this should be super edge case-y but timestamp would be way better